src/core/config.c

```diff
@@ -198,38 +198,38 @@ bool mCoreConfigSaveVFile(const struct mCoreConfig* config, struct VFile* vf) {
 	return ConfigurationWriteVFile(&config->configTable, vf);
 }
 
-void mCoreConfigMakePortable(const struct mCoreConfig* config) {
+void mCoreConfigMakePortable(const struct mCoreConfig* config, const char* path) {
 	struct VFile* portable = NULL;
 	char out[PATH_MAX];
-	mCoreConfigPortablePath(out, sizeof(out));
+	mCoreConfigPortableIniPath(out, sizeof(out));
 	if (!out[0]) {
 		// Cannot be made portable
 		return;
 	}
 	portable = VFileOpen(out, O_WRONLY | O_CREAT);
 	if (portable) {
+		if (path && path[0]) {
+			struct Configuration portableConfig;
+			ConfigurationInit(&portableConfig);
+			ConfigurationSetValue(&portableConfig, "portable", "path", path);
+			ConfigurationWriteVFile(&portableConfig, portable);
+			ConfigurationDeinit(&portableConfig);
+		} else {
+			// For an empty path, reset the file to also be empty.
+			portable->truncate(portable, 0);
+		}
```

Comment:
For future proofing, I'd prefer if you loaded the old file and just update or delete that key.